tests/bears/BearTest.py

```diff
@@ -537,6 +559,107 @@ def test_is_debugged(self):
         section.append(Setting('debug_bears', 'abc, xyz'))
         self.assertEqual(_is_debugged(uut), False)
 
+    @patch('cProfile.Profile.dump_stats')
+    def test_profiler_with_no_directory_exists(self, dump_stats):
+        args = ()
+        kwargs = {}
+        section = Section('name')
+        section.append(Setting('profile', 'tRuE'))
+        bear = TestTwoBear(section, self.queue)
+        self.assertEqual(bear.run_bear_from_section(args, kwargs), [1, 2, 3])
+        dump_stats.assert_called_once_with('name_TestTwoBear.prof')
+        section.append(Setting('profile', 'abc'))
+        bear = TestTwoBear(section, self.queue)
+        self.assertEqual(bear.run_bear_from_section(args, kwargs), [1, 2, 3])
+        dump_stats.assert_called_with(os.path.join(
+            bear.profile, 'name_TestTwoBear.prof'))
+        os.rmdir('abc')
+        section.append(Setting('profile', '1'))
+        bear = TestThreeBear(section, self.queue)
+        self.assertIsNone(bear.run_bear_from_section(args, kwargs))
+        dump_stats.assert_called_with('name_TestThreeBear.prof')
+
+    @patch('cProfile.Profile.dump_stats')
+    def test_profiler_with_directory_exists(self, dump_stats):
+        with TemporaryDirectory() as temp_dir:
+            args = ()
```

Comment:
the other three methods declare `args = ()` and `kwargs = {}` outside the `with`.  why not here too for consistency?

Reply:
Yeah I can do that :+1: Its not like something ambiguous :)